Add barrier 8 (big data) to barriers.md in reproducibility chapter (#703)
alan-turing-institute/the-turing-way: book/content/reproducibility/04/barriers.md

*replace this text with the content of barrier 7*

## Big data and complex computational infrastructure

Big data is conceptualised in different ways by different researchers. 

> The traditional understanding of big data focuses on complexity, variety of data sources, volume and velocity of the data. 

Reviewer: ```suggestion
"Big" data may be complex, come from a variety of data sources, is large in storage volume and/or be streamed at very high temporal resolution. 
```

> It is extremely difficult to control to ensure consistency of data between the experiments. 

Reviewer: ```suggestion
Although there are ways to set random seeds and take snapshots of a dataset at a particular moment in time, it can be difficult to have identical data across different runs of an analysis pipeline.
This is particularly relevant in the context of tools for parallel computing.
```

> In some cases, when data is sampled from a larger collection of data consistency cannot be guaranteed.

Reviewer: Tagging @crowcroft because I remember him bringing this up years ago! Jon - do you have any good links to data sources to add to the flight tracking or internet traffic parts of this sentence?

```suggestion
For example, streaming data such as flight tracking or internet traffic can not be stored and must be processed in real time.
```

> A large variety of tools and products with a constantly changing ecosystem of supported technologies is available, meaning reproducing results in the future is highly dependent on the available technology at the time. 

Reviewer: ```suggestion
A more common challenge for "big data" researchers is the variability of software performance across operating systems and how quickly the tools change over time.
An almost constantly changing ecosystem of data science technologies is available, which means reproducing results in the future is highly variable and dependent on using perfectly backwards compatible tools as they develop. 
```

> Some tools require in depth technical skills which are not widely available to data scientists. 

Reviewer: I've just moved this section around a little, I hope that's ok?

Do you know of a link to the point about setting a random seed in tensorflow limiting parallelisation options? That's a great point!
```suggestion
Very often the results of statistical tests will vary depending on the configuration of the infrastructure that was used in each of the experiments, making it very hard to independently reproduce a result.
Experiments are often dependent on random initialisation for iterative algorithms and not all software includes the ability to fix a pseudorandom number without limiting parallelisation capabilities (for example  e.g. in Tensorflow).
These tools can require in depth technical skills which are not widely available to data scientists.
```

The Hadoop framework, for instance, is extremely complex to deploy data science experiments without strong software and hardware engineering knowledge. 

> Combined with the need for high performance computing, the barrier to access big data processing is even greater. 

Reviewer: ```suggestion
Even "standard" high performance computing, can be difficult to set up to be perfectly reproducible, particularly across different cloud computing providers or institutional configurations.
```

> Very often the results of statistical tests will vary depending on the configuration of the infrastructure that was used in each of the experiments, therefore reproducibility is difficult to achieve (experiments are often dependent on random initialisation for iterative algorithms, fixing a pseudorandom number will result in limiting parallelisation capabilities, e.g. in Tensorflow).

Reviewer: Moved to the start of the paragraph.
```suggestion
```
